std/algorithm/searching.d

```diff
@@ -799,15 +802,33 @@ if (isForwardRange!R
                   haystack.popFront();
             }
         }
+        //Because of @@@8804@@@: Avoids both "unreachable code" or "no return statement"
```

Comment:
Isn't this simply because there's no return statement in the `else` branch above?